src/interface/sparse_pack.hpp

```diff
@@ -200,27 +252,32 @@ class SparsePack : public SparsePackBase {
   KOKKOS_INLINE_FUNCTION int GetLowerBound(const int b) const { return 0; }
 
   KOKKOS_INLINE_FUNCTION int GetUpperBound(const int b) const {
+    PARTHENON_DEBUG_REQUIRE(!flat_, "Bounds not valid for flat packs");
```

Comment:
Are they not valid if `b==0`? I thought they were.

Reply:
Ok, after reading below I see they are not valid. I think we should make it so that they are valid, and maybe add a default `const int b=0` here so that `GetLowerBound()` and `GetUpperBound()` work for sparse packs (or maybe make them separate functions that have a check if `flat_==true`). I think this is clearer than exposing the pack size through  `GetMaxNumberOfVars()`.

Reply:
Maybe better to just call them `GetFlatLowerBound()` and `GetFlatUpperBound()`.

Reply:
I'm going to make them valid and just re-use the API. I'm not sure if we should come up with new or more generic names, I just chose not to for now. We can revisit that later maybe.